bgpd/bgpd.c

```diff
@@ -671,6 +671,7 @@ int bgp_confederation_peers_add(struct bgp *bgp, as_t as)
 	if (bgp_config_check(bgp, BGP_CONFIG_CONFEDERATION)) {
 		for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer)) {
 			if (peer->as == as) {
+				(void)peer_sort(peer);
```

Comment:
I still have one question tho: Can't we update sort outside this check? I'm curious about other _upcoming_ bugs (maybe or not).

Reply:
Yes, we can update peer's sort outside the check above which effectively means for every configured peer. But we need to update only the affected peer whose AS number was added to a confederation.